src/parser/core/modules/Statuses.js

```diff
@@ -0,0 +1,174 @@
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {ItemGroup, Item} from './Timeline'
+import {getDataBy} from 'data'
+import React from 'react'
+
+const STATUS_APPLY_ON_PARTY_THRESHOLD_MILLISECONDS = 2 * 1000
+
+// Track statuses applied by actions
+export default class Statuses extends Module {
+	static handle = 'statuses'
+	static dependencies = [
+		'timeline',
+		'cooldowns',
+		'gcd',
+	]
+
+	static statusesStackMapping = {}
+
+	_statuses = {}
+	_groups = {}
+	_statusToActionMap = {}
+	_actionToMergeNameMap = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const ids = [this.parser.player.id, ...this.parser.player.pets]
+		const byFilter = {by: ids}
+
+		this.addHook('complete', this._onComplete)
+		this.addHook(['applybuff', 'applydebuff'], byFilter, this._onApply)
+		this.addHook(['refreshdebuff', 'refreshbuff'], byFilter, this._onRefresh)
+		this.addHook(['removebuff', 'removedebuff'], byFilter, this._onRemove)
+
+		this.cooldowns.constructor.cooldownOrder.forEach(it => {
+			if (typeof it === 'object' && it.merge) {
+				it.actions.forEach(ac => {
+					this._actionToMergeNameMap[ac] = it.name
+				})
+			}
+		})
+
+		// Map statuses to actions
+		Object.values(ACTIONS).forEach(ac => {
+			if (ac.statusesApplied) {
+				ac.statusesApplied.forEach(st => {
+					if (st) {
+						this._statusToActionMap[st.id] = ac.id
```

Comment:
medium; It looks like the only place the value of the map is used, you're pulling up the action data with `getDataBy`. Object references are cheap in JS, you'd be better off just setting the entire action object as the value in the map here, and skipping the `getDataBy` call later.

Reply:
done